sonar-plugin-api/src/main/java/org/sonar/api/server/authentication/UserIdentity.java

```diff
@@ -131,23 +189,35 @@ public UserIdentity build() {
       return new UserIdentity(this);
     }
 
-    private static void validateProviderLogin(String providerLogin){
+    private static void validateProviderLogin(String providerLogin) {
       checkArgument(isNotBlank(providerLogin), "Provider login must not be blank");
       checkArgument(providerLogin.length() <= 255, "Provider login size is incorrect (maximum 255 characters)");
     }
 
-    private static void validateLogin(String login){
+    private static void validateLogin(String login) {
       checkArgument(isNotBlank(login), "User login must not be blank");
       checkArgument(login.length() <= 255 && login.length() >= 3, "User login size is incorrect (Between 3 and 255 characters)");
     }
 
-    private static void validateName(String name){
+    private static void validateName(String name) {
       checkArgument(isNotBlank(name), "User name must not be blank");
       checkArgument(name.length() <= 200, "User name size is too big (200 characters max)");
     }
 
-    private static void validateEmail(@Nullable String email){
+    private static void validateEmail(@Nullable String email) {
       checkArgument(email == null || email.length() <= 100, "User email size is too big (100 characters max)");
     }
   }
+
+  private enum ValidateGroupName implements Predicate<String> {
+    INSTANCE;
+
+    @Override
+    public boolean apply(@Nonnull String input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this "Nonnull" annotation to honor the overridden method's contract. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)